pkg/controller/utils/infinispan/configuration/xsite.go

```diff
@@ -170,7 +170,7 @@ func getNodePortServiceHostPort(service *corev1.Service, k *util.Kubernetes, log
 	workerList := &corev1.NodeList{}
 
 	//select workers only. AFAIK, only the workers have the proxy to the pods
-	labelSelector := labels.SelectorFromSet(map[string]string{"node-role.kubernetes.io/worker": ""})
+	labelSelector := labels.SelectorFromSet(map[string]string{"node-role.kubernetes.io/worker": "true"})
```

Comment:
I'm wondering if we really need this selector at all. It seems that also "master" nodes expose NodePort service

Reply:
This need to be tested

Reply:
I've tested on OCP and minikube and all the nodes serve NodePort request

Reply:
Oh, good, let's remove this label selector. We also need to backport this PR in 2.0.x also

Reply:
@rigazilla you should also validate, that you don't have a cluster where `worker` and `master` node are combine on one host.

Reply:
yeah:
OCP has 1 master, 2 worker
minikube only 1 master   (node-role worker)

looking at the NodePort docu, I didn't find any reference to the master/worker distinction

Reply:
Yes, sure, @rigazilla 
I've also validated, it's true that NodePort is exposed on all nodes. But this will work for happy-path only. What happens, if we choose a `master` node host, make configuration and then that node will go down? Nothing changes in Pods and operator reconciliation happens, it's just a `master` , no Pods scheduling, etc. I think we need to choose a `worker` node with at least one scheduled Infinispan cluster Pod. 
P.S. It's just an idea, and this unhappy path need to be validated, but the precise choice get for us a more chances for success.

Reply:
@dmvolod I can see the problem you want to address, but I dont think there's an easy solution atm: even if the reconciliation loop happens on the cluster where node goes down we need to fix the configuration on the other cluster.
also I'm not sure we can change the jgroups configuration at runtime.
I think that a proper solution to this problem would be to have more than 1 coordinator per cluster or maybe rely on Skupper recovery features. #385